That is another question :). Build is failing on circleci, but it is actually dmd failing. Not sure why, I see no help in the output:

```
++ c++ -m64 -c -o test_results/compilable/issue15574/square.c.o test_results/compilable/issue15574/square.c
++ ar rcs test_results/compilable/issue15574/libcsquare.a test_results/compilable/issue15574/square.c.o
++ ../generated/linux/release/64/dmd -m64 -lib -oftest_results/compilable/issue15574/libdsquare.a test_results/compilable/issue15574/square.d
++ ../generated/linux/release/64/dmd -m64 -oftest_results/compilable/issue15574/app test_results/compilable/issue15574/app.d -Itest_results/compilable/issue15574 test_results/compilable/issue15574/libdsquare.a -L-Ltest_results/compilable/issue15574 -L-lcsquare
+ finish 1
+ set +x
Makefile:133: recipe for target 'test_results/compilable/issue15574.sh.out' failed
```

--- a/test/compilable/issue15574.sh
+++ b/test/compilable/issue15574.sh
@@ -19,6 +19,10 @@ APP=$TEST_DIR/app${EXE}
 mkdir -p $TEST_DIR
 
 cat >$C_FILE <<EOF
+#ifdef __cplusplus
+  extern "C" int square(int x);
+#endif
+
 int square(int x) { return x*x; }
 EOF
 
@@ -44,7 +48,7 @@ int main() {
 }
 EOF
 
-cc -m${MODEL} -c -o ${C_FILE}${OBJ} $C_FILE
+${CC} -m${MODEL} -c -o ${C_FILE}${OBJ} $C_FILE
 ar rcs ${C_LIB} ${C_FILE}${OBJ}
 
 ${DMD} -m${MODEL} -lib -of${D_LIB} ${D_FILE}

